### PR TITLE
Enable compiling the test projects against the configuration

### PR DESCRIPTION
**Pull request by tarekgh**
The tests were compiled using the vertical configuration and not the best configuration picked from Configurations.props. this change is fixing that and have all test projects configured accordingly.
This change ensuring netcoreapp runs successfully and we'll have another rounds later for other configurations like netfx, UAP...etc.

### CODE REVIEW

**Comment by tarekgh** [comment]
@ericstj @weshaggard could you please have a look? 

CC @joperezr

File: dir.props [hunk 1]
```diff
@@ -252,8 +252,6 @@
     <BuildConfigurationRefPath Condition="$(_bc_TargetGroup.EndsWith('aot'))">$(RefRootPath)$(_bc_TargetGroup.TrimEnd('aot'))/</BuildConfigurationRefPath>
     <RefPath>$(RefRootPath)$(TargetGroup)/</RefPath>
     <RefPath Condition="$(TargetGroup.EndsWith('aot'))">$(RefRootPath)$(TargetGroup.TrimEnd('aot'))/</RefPath>
-    <!-- tests don't yet set appropriate configurations -->
-    <RefPath Condition="'$(IsTestProject)' == 'true'">$(BuildConfigurationRefPath)</RefPath>
```
Comment (weshaggard) [inline dir.props]: Does this mean you fixed all the configurations in the test projects and not just the netfx configurations?

**Comment by weshaggard** [comment]
As part of these configuration changes can you please update the project and solution configs? Just run `msbuild build.proj /t:UpdateVSConfigurations` from the root.

File: dir.props (hunk 1 above)
Comment (tarekgh) [inline dir.props]: no this is just getting the test projects compile against the configuration with netcoreapp. we'll need to go and fix the projects more for other verticals (netfx, UAP...etc.)

**Comment by tarekgh** [comment]
@weshaggard 

> As part of these configuration changes can you please update the project and solution configs? Just run msbuild build.proj /t:UpdateVSConfigurations from the root.

I did that.

**Comment by weshaggard** [comment]
@tarekgh did you have my latest changes that merged a little bit ago? I ask because I don't see any sln files updated but I believe some of your changes might cause them to update (although I didn't look very closely).

File: src/System.Collections.Immutable/tests/Configurations.props [hunk 2]
```diff
@@ -3,6 +3,8 @@
   <PropertyGroup>
     <BuildConfigurations>
       netstandard1.3;
+      netstandard;
+      netcoreapp;
```
Comment (weshaggard) [inline src/System.Collections.Immutable/tests/Configurations.props]: I would definitely expect this to caused a solution file to be updated when you run an update. It should now prefer the netcoreapp configuration for this test project.

**Comment by tarekgh** [comment]
@weshaggard 

msbuild build.proj /t:UpdateVSConfigurations didn't update any sln files. only csproj files get updated. if you expect sln files get updated then this means the target has a problem

**Comment by tarekgh** [comment]
@weshaggard I have updated the sln files too.

File: src/Common/tests/Configurations.props [hunk 3]
```diff
@@ -1,9 +1,11 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
       netstandard1.3-Unix;
```
Comment (weshaggard) [inline src/Common/tests/Configurations.props]: For the test projects I think we should go a head and remove all the non-version-less configurations. We might need them for some of the src/ref projects but we don't need them for the test projects.

Comment (tarekgh) [inline src/Common/tests/Configurations.props]: I'll do this clean up later in another PR.

**Pull request merged**